packages/bot-skeleton/src/services/api/appId.js

```diff
@@ -1,35 +1,177 @@
-};
+    const getSocketUrl = () =>
+        `wss://${getSocketURL()}/websockets/v3?app_id=${getAppId()}&l=${getLanguage()}&brand=${website_name.toLowerCase()}`;
 
-const getDomainAppId = () => AppIds[document.location.hostname.replace(/^www./, '')];
+    const isReady = () => hasReadyState(1);
 
-const getDefaultEndpoint = () => ({
-    url: isRealAccount() ? 'green.binaryws.com' : 'blue.binaryws.com',
-    appId: localStorage.getItem('config.default_app_id') || getDomainAppId() || 19111,
-});
+    const isClose = () => !binary_socket || hasReadyState(2, 3);
 
-const getServerAddressFallback = () => getCustomEndpoint().url || getDefaultEndpoint().url;
+    const close = () => {
+        binary_socket.close();
+    };
 
-const getAppIdFallback = () => getCustomEndpoint().appId || getDefaultEndpoint().appId;
+    const hasReadyState = (...states) => binary_socket && states.some(s => binary_socket.readyState === s);
 
-const getWebSocketURL = () => `wss://${getServerAddressFallback()}/websockets/v3`;
+    const init = ({ options }) => {
+        if (typeof options === 'object' && config !== options) {
+            config = options;
+        }
+    };
 
-export const generateLiveApiInstance = () =>
-    new LiveApi({
-        apiUrl: getWebSocketURL(),
-        language: getLanguage().toUpperCase(),
-        appId: getAppIdFallback(),
+    const openNewConnection = () => {
+        if (wrong_app_id === getAppId()) return;
+
+        if (isClose()) {
+            is_disconnect_called = false;
+            binary_socket = new WebSocket(getSocketUrl());
+            deriv_api = new DerivAPIBasic({
+                connection: binary_socket,
+            });
+        }
+
+        deriv_api.onOpen().subscribe(() => {
+            wait('website_status');
+
+            if (typeof config.onOpen === 'function') {
+                config.onOpen(isReady());
+            }
+
+            if (typeof config.onReconnect === 'function' && is_connected_before) {
+                config.onReconnect();
+            }
+
+            if (!is_connected_before) {
+                is_connected_before = true;
+            }
+        });
+
+        deriv_api.onMessage().subscribe(({ data: response }) => {
+            const msg_type = response.msg_type;
+            State.set(['response', msg_type], cloneObject(response));
+
+            if (getPropertyValue(response, ['error', 'code']) === 'InvalidAppID') {
+                wrong_app_id = getAppId();
+            }
+
+            if (typeof config.onMessage === 'function') {
+                config.onMessage(response);
+            }
+        });
+
+        deriv_api.onClose().subscribe(() => {
+            if (wrong_app_id !== getAppId() && typeof config.onDisconnect === 'function' && !is_disconnect_called) {
+                config.onDisconnect();
+                is_disconnect_called = true;
+            }
+        });
+    };
+
+    const availability = status => {
+        if (typeof status !== 'undefined') {
```

Comment:
Just curious, why're you using `typeof` here? 
You've written `typeof status !== 'undefined'` instead of `status !== 'undefined'` ?